components/Common/Sidebar/SidebarGroup/index.tsx

```diff
@@ -0,0 +1,43 @@
+import type { FC } from 'react';
+
+import SidebarItem from '@/components/Common/Sidebar/SidebarItem';
+
+import styles from './index.module.css';
+import type { ActiveItem, SidebarGroupType } from '..';
+
+type SideBarGroupProps = SidebarGroupType & {
+  activeItem?: ActiveItem;
+  setActiveItem: (item: ActiveItem) => void;
+};
+
+const SidebarGroup: FC<SideBarGroupProps> = ({
+  groupName,
+  items,
+  activeItem,
+  setActiveItem,
+}) => {
+  const handleSideBarItemClick = (title: string) => {
+    setActiveItem({ groupName, title });
+  };
+
+  return (
+    <section className={styles.group}>
+      <label className={styles.groupName}>{groupName}</label>
+      <ul className={styles.itemList}>
+        {items.map(item => (
+          <SidebarItem
+            key={item.title}
+            {...item}
+            isActive={
+              activeItem?.groupName === groupName &&
+              activeItem?.title === item.title
+            }
+            onClick={handleSideBarItemClick}
+          />
+        ))}
```

Comment:
```suggestion
        {items.map(({ title, url, isActive }) => (
          <SidebarItem
            key={title}
            url={url}
            title={title}
            isActive={
              isActive || (title === activeItem?.title && url === activeItem?.url)
            }
          />
        ))}
```

You can use destructuring into the `item`. IMO, need to remove `{...item}` and send the props directly